src/main/scala/org/deepdive/ddlog/DeepDiveLogParser.scala

```diff
@@ -14,6 +14,7 @@ case class Variable(varName : String, relName : String, index : Int )
 
 sealed trait Expr
 case class VarExpr(name: String) extends Expr
+case class ArrayExpr(name: String, index: Int) extends Expr
```

Comment:
Let's make this `ArrayElementExpr(array: Expr, index: Expr)`, not just a variable with integers.